### ISSUE TITLE
Provider produced inconsistent result after apply with azurerm_sentinel_alert_rule_scheduled resource

### ISSUE THREAD

**Issue by davidlife52** [hashicorp/terraform-provider-azurerm#21231]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

I need to deploy 50+ sentinel scheduled alert rules. I have written a terraform script which JSON decodes every file (ARM template files) from a given directory and fills the necessary values in azurerm_sentinel_alert_rule_scheduled resource. 

After terraform apply, I get the following error:

![image](https://user-images.githubusercontent.com/47814022/229117090-cdeb04d0-c58a-4dcc-ab76-c0568f17dab6.png)

After retrying a couple of times, on average I get this error for 5 alert rules out of 50. Every time the alert rules that produces the error is different. The ones that do not produce the error, always deploys successfully. I've tried importing the rule directly to sentinel from the arm template that I parse, by hand, and no problems occur.

If the error was produced by the same rules, I'd suspect a typo somewhere that I made, but every time errors get produced by the different rules and the rules that threw the error before, deploys successfully.

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

v1.4.4

### AzureRM Provider Version

v3.50.0

### Affected Resource(s)/Data Source(s)

azurerm_sentinel_alert_rule_scheduled

### Terraform Configuration Files

```hcl
locals {

  scheduled_alert_files = fileset(path.module, "ScheduledRules/*.json")
  scheduled_alert_rules = { for file in local.scheduled_alert_files : file => { data = jsondecode(file("${path.module}/${file}")) }}
}
resource "azurerm_sentinel_alert_rule_scheduled" "scheduled_alert_rules" {
  for_each                    = local.scheduled_alert_rules
  
  name                        = trim(split("/", each.value.data.resources[0].id)[length(split("/", each.value.data.resources[0].id)) - 1], "')]")
  log_analytics_workspace_id  = var.analytics_workspace_id
  display_name                = each.value.data.resources[0].properties.displayName
  severity                    = each.value.data.resources[0].properties.severity
  query                       = each.value.data.resources[0].properties.query

  alert_rule_template_guid    = each.value.data.resources[0].properties.alertRuleTemplateName
  alert_rule_template_version = each.value.data.resources[0].properties.templateVersion
  custom_details              = each.value.data.resources[0].properties.customDetails
  description                 = each.value.data.resources[0].properties.description
  enabled                     = false

  dynamic "entity_mapping" {
    for_each = each.value.data.resources[0].properties.entityMappings != null ? each.value.data.resources[0].properties.entityMappings : []
    content {
      entity_type = entity_mapping.value.entityType
      dynamic "field_mapping"{
        for_each = entity_mapping.value.fieldMappings
        content{
          identifier = field_mapping.value.identifier
          column_name = field_mapping.value.columnName
        }
      }
    }
  }
  event_grouping {
    aggregation_method = each.value.data.resources[0].properties.eventGroupingSettings.aggregationKind
  }
  incident_configuration {
    create_incident = each.value.data.resources[0].properties.incidentConfiguration.createIncident
    grouping {
      enabled = each.value.data.resources[0].properties.incidentConfiguration.groupingConfiguration.enabled
      lookback_duration = each.value.data.resources[0].properties.incidentConfiguration.groupingConfiguration.lookbackDuration
      reopen_closed_incidents = each.value.data.resources[0].properties.incidentConfiguration.groupingConfiguration.reopenClosedIncident
      entity_matching_method = each.value.data.resources[0].properties.incidentConfiguration.groupingConfiguration.matchingMethod
      group_by_entities = each.value.data.resources[0].properties.incidentConfiguration.groupingConfiguration.groupByEntities
      group_by_alert_details = each.value.data.resources[0].properties.incidentConfiguration.groupingConfiguration.groupByAlertDetails
      group_by_custom_details = each.value.data.resources[0].properties.incidentConfiguration.groupingConfiguration.groupByCustomDetails
    }
  }
  query_frequency = each.value.data.resources[0].properties.queryFrequency
  query_period = each.value.data.resources[0].properties.queryPeriod
  suppression_duration = each.value.data.resources[0].properties.suppressionDuration
  suppression_enabled = each.value.data.resources[0].properties.suppressionEnabled
  dynamic "sentinel_entity_mapping" {
    for_each = each.value.data.resources[0].properties.sentinelEntitiesMappings != null ? each.value.data.resources[0].properties.sentinelEntitiesMappings : []
    content{
      column_name = sentinel_entity_mapping.value.fieldMappings[0].columnName
    }
  }

  tactics = each.value.data.resources[0].properties.tactics
  techniques = each.value.data.resources[0].properties.techniques
  trigger_operator = each.value.data.resources[0].properties.triggerOperator
  trigger_threshold = each.value.data.resources[0].properties.triggerThreshold
}
```

### Debug Output/Panic Output

```shell
│ Error: Provider produced inconsistent result after apply
│ 
│ When applying changes to
│ module.tdrs[0].module.tdrsOption1.azurerm_sentinel_alert_rule_scheduled.scheduled_alert_rules["ScheduledRules/alert_rule_file.json"],
│ provider "provider[\"registry.terraform.io/hashicorp/azurerm\"]" produced
│ an unexpected new value: Root resource was present, but now absent.
│ 
│ This is a bug in the provider, which should be reported in the provider's
│ own issue tracker.
```

### Expected Behaviour

Creation completed after *s; No errors occurred.

**Comment by ziyeqf** [comment]
Hi @davidlife52, thanks for reporting.

Could you please provide a log file which helps me to dig into it?

To do that you can set the environment variable `TF_LOG` to `DEBUG` and `TF_LOG_PATH` to a file path. And remember to remove you credential info in the log file, replace them with some other strings before upload the log file here. :)

Thanks!

**Comment by davidlife52** [comment]
HI @ziyeqf 

I've been trying to get the error in debug mode, but it doesn't appear any more.

After creating log analytics and sentinel resources with a new name I got an error, that sentinel is not onboarded. That's when I've noticed that azurerm_sentinel_alert_rule_scheduled resource requires log analytics workspace and not sentinel, so terraform never saw a dependency between two of them. So I added the dependency explicitly.

I think that might been the case, or Azure did something on their end that had an effect on the error, because I was rerunning terraform apply then destroy multiple times to get the error before explicitly defining the dependency, but I haven't come across the error Today.  The error regarding sentinel not being onboarded yet, show up only if I create sentinel and log analytics resources with a specific name for the first time, if I delete everything and then try to create the resources with the same name again, I do not get the error anymore.

Thanks for taking time to look at this issue!
